Author: Wolfgang Rohdewald <wolfgang@rohdewald.de>
Description: this continues the fix from r1301756
 .
 those two commits are needed for ubuntu quantal which is using the latest
 sqlite / qtsql packages
Applied-Upstream: 4.10, commit:1301764

--- a/kajongg/src/scoringengine.py
+++ b/kajongg/src/scoringengine.py
@@ -261,10 +261,12 @@ class Ruleset(object):
         if used is not None:
             self.__used = used
         records = Query("select max(id)+1 from %s" % self.__rulesetTable()).records
-        try:
-            return int(records[0][0])
-        except ValueError:
-            return 1
+        if records and records[0] and records[0][0]:
+            try:
+                return int(records[0][0])
+            except ValueError:
+                return 1
+        return 1
 
     @staticmethod
     def nameIsDuplicate(name):
